client/src/js/components/bhAccountSelect.js

```diff
@@ -29,19 +29,15 @@ function AccountSelectController(Accounts, AppCache, $timeout, bhConstants, $sco
   var hasCachedAccounts = false;
   var cache = new AppCache('bhAccountSelect');
 
-  // cache accounts locally for three seconds
-  var CACHE_TIMEOUT = 3000;
-
   // fired at the beginning of the account select
   $ctrl.$onInit = function $onInit() {
 
-    $ctrl.classe = $ctrl.accountTypeId || '';
     // cache the title account ID for convenience
     $ctrl.TITLE_ACCOUNT_ID = bhConstants.accounts.TITLE;
 
     // To filter accounts by class of accounts
-    $ctrl.classe = $ctrl.classe || '';
-
+    $ctrl.classe = $ctrl.classe || undefined;
```

Comment:
If both of these are optional, they should be set to `undefined` by default.  What happens if you remove these two lines?